src/main/java/seedu/address/logic/commands/archive/RemoveCommand.java

```diff
@@ -0,0 +1,71 @@
+package seedu.address.logic.commands.archive;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.model.Model.PREDICATE_SHOW_ARCHIVED_PERSONS;
+
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+public class RemoveCommand extends Command {
+    public static final String COMMAND_WORD = "archive remove";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Removes a person from the archive without deleting the person.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_REMOVE_SUCCESS = "Removed Person from archive: %1$s. "
+            + "This Person will appear on your contact list again.";
+
+    private final Index targetIndex;
+
+    public RemoveCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getSortedPersonList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToArchive = lastShownList.get(targetIndex.getZeroBased());
+
+        if (!personToArchive.isArchived()) {
+            throw new CommandException(Messages.MESSAGE_UNARCHIVE_INVALIID_LIST);
+        }
+
+        Person archivedPerson = new Person(
+                personToArchive.getId(),
+                personToArchive.getName(),
+                personToArchive.getPhone(),
+                personToArchive.getEmail(),
+                personToArchive.getAddress(),
+                personToArchive.getTags(),
+                personToArchive.getRemark(),
+                false,
+                personToArchive.getTotalSalesAmount()
+        );
+        model.setPerson(personToArchive, archivedPerson);
```

Comment:
I think the naming is quite confusing for the command archive remove.Would naming  personToArchive to archivedPerson and archivedPerson be named to newUnarchivedPerson or something be clearer?

Reply:
Yeah did too much Ctrl+C Ctrl+V. Will fix the naming asap. Also what would be a better name for the command?